ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -88,32 +93,158 @@ tests = testGroup "OnDisk" [
     ]
 
 {-------------------------------------------------------------------------------
-  Auxiliary functions for working with TestBlock
+  TestBlock
+-------------------------------------------------------------------------------}
+
+type TestBlock = TestBlockWith Tx
+
+-- | Simple block transaction
+--
+data Tx = Tx { consumed :: Token, produced :: (Token, TValue) }
+  deriving stock (Show, Eq, Ord, Generic)
+  deriving anyclass (Serialise, NoThunks, ToExpr)
+
+newtype Token = Token (Point TestBlock)
```

Comment:
Hmm, I find making an analogy to `TxIn` and `TxOut` harder than I thought. In the UTxO model, TxIn and TxOut have different types, whereas I'm using `Token` for modelling both what a transaction consumes and produces. I don't think we could make such an analogy here. I could comment that the field `consumed` would be analogous to a transaction input, and `produced` to a transaction output, but I don't know if that is what you had in mind.

![image](https://user-images.githubusercontent.com/175315/154047354-8fb51b09-0079-4c2c-bf56-8e3cd00949c3.png)

Reply:
This was the mapping I envisioned:

```
TxId: Token
TxIn: (Token, ())
TxOut: TValue
```

In reality `TxIn = (TxIn, Nat)` since real txs can have 0-to-many outputs. This payload's txs always have exactly one output, hence the `()` above.

You're right that `TValue` is most analogous to `Coin`, and that `TxOut` has more than `Coin`. But I think "analogy" is loose enough that that's OK.

Reply:
So concretely, should I simply write:

```haskell
-- | This is analogous to @TxIn@
newtype Token = ...

-- | This is analogous to @TxOut@
newtype TValue = ...
```

Reply:
That'd be fine, yep. Here's what I would probably write.

```
-- | This is analogous to @TxId@: it's how we identify what's in the table
--
-- It's also analogous to @TxIn@, since we trivially only have one output per 'Tx'.
newtype Token = ...

-- | This is analogous to @TxOut@: it's what the table maps 'Token's to
newtype TValue = ...
```

Reply:
Your latest commit looks fantastic! Resolving this Conversation.